### PR TITLE
wgsl: Implement abstract shift tests (<< and >>)

### PR DESCRIPTION
**Pull request by zoddicus**
Issue #1631

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [ ] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [ ] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts [hunk 1]
```diff
@@ -60,14 +61,12 @@ function isValidConstShiftRight(e1: ScalarValue, e2: number) {
...
 }
 
 // Returns all cases of shifting e1 left by [0,63]. If `isConst` is true, cases that are
 // invalid for const eval are not returned.
-function generateShiftLeftCases(e1: ScalarValue, isConst: boolean): Case[] {
-  // This will need to change when AbstractInts are added in
+function generateShiftLeftConcreteCases(e1: ScalarValue, isConst: boolean): Case[] {
   assert(typeof e1.value === 'number');
 
   const bitwidth = e1.type.size * 8;
```
Comment (amaiorano) [inline src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts:72]: nit: can we make the for loop in this function consistent with the one in `generateShiftLeftAbstractCases` : `for (let e2 = 0; e2 < 64; ++e2) {` ?

(GitHub is weird - I can't add comments on lines you did not modify)

File: src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts [hunk 2]
```diff
@@ -83,12 +82,25 @@ function generateShiftLeftCases(e1: ScalarValue, isConst: boolean): Case[] {
...
+
 // Returns all cases of shifting e1 right by [0,63]. If `is_const` is true, cases that are
 // invalid for const eval are not returned.
-function generateShiftRightCases(e1: ScalarValue, isConst: boolean): Case[] {
-  // This will need to change when AbstractInts are added in
+function generateShiftRightConcreteCases(e1: ScalarValue, isConst: boolean): Case[] {
   assert(typeof e1.value === 'number');
-
   const cases: Case[] = [];
   for (let shift = 0; shift < 64; ++shift) {
```
Comment (amaiorano) [inline src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts:105]: nit: same as above, consistent loop

**Review by amaiorano — APPROVED** [review]
Apart from minor nits, and a question, this LGTM!

File: src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts:72]: Done

File: src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts (hunk 2 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts:105]: Done

File: src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts [hunk 3]
```diff
@@ -109,7 +121,126 @@ function generateShiftRightCases(e1: ScalarValue, isConst: boolean): Case[] {
...
+function generateShiftRightAbstractCases(e1: ScalarValue): Case[] {
+  assert(typeof e1.value === 'bigint');
+  const cases: Case[] = [];
+  // Shifting within 64 bits
+  for (let e2 = 0; e2 < 64; ++e2) {
+    const expected: bigint = e1.value >> BigInt(e2);
+    cases.push({ input: [e1, u32(e2)], expected: abstractInt(expected) });
+  }
+  // Always filled values
+  for (let e2 = 64; e2 < 1025; e2 *= 2) {
```
[the remark this answers is not in the document]
Comment (zoddicus) [reply src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts:134]: I think I muddled two approaches to this together. Removed the `+ 1` so 64 is covered.

**Comment by zoddicus** [comment]
The right shift overflow tests being broken for Dawn/Tint is a reported issue, https://issues.chromium.org/issues/339041265

**Pull request merged**